SourceCode/src/light.cpp

```diff
@@ -0,0 +1,29 @@
+#include "light.hpp"
+
+Light::Light(float intensity, const cm::Vec3& position) : intensity(intensity), position(position)
+{}
+
+cm::Vec3 Light::get_contribution(const cm::Vec3& pos, const cm::Vec3& normal)
+{
+  cm::Vec3 contribution(0.0);
+  const float distance = cm::length(position - pos);
+  contribution = cm::Vec3((intensity * std::max(0.0f, cm::dot(position - pos, normal))) / (distance * distance));
```

Comment:
You're not taking into account the imaginary light sphere radius, it's not wrong but it could lead to different results compared to the final images in the courses slides.

Reply:
@VladislavVulchevChaos I thought a point light source had zero extent. So, how can I consider an imaginary light sphere radius? Could you provide me some more detail?

Reply:
nevermind, I got it. We consider the intensity of the light as the total intensity emitted by the light. Thus, we need to 'distribute' this intensity over the spread of the light from its origin.